theano/tensor/tests/test_basic.py

```diff
@@ -1683,7 +1718,7 @@ def test_py_c_match():
     expected_gamma = scipy.special.gamma
     expected_gammaln = scipy.special.gammaln
     expected_psi = scipy.special.psi
-    expected_chi2sf = lambda x, df: scipy.stats.chi2.sf(x, df).astype(x.dtype)
+    expected_chi2sf = lambda x, df: scipy.stats.chi2.sf(x, df)
```

Comment:
I don't know why previously the results were cast to ```x.dtype```. I removed it simply. Please check this out.